src/GovUk.Education.ExploreEducationStatistics.Content.Model/Extensions/ReleaseAmendmentExtensions.cs

```diff
@@ -0,0 +1,144 @@
+{
+    public static class ReleaseAmendmentExtensions
+    {
+        private static readonly string FilterPattern = string.Join(
+            '|',
+            // Strips out anything that looks like a comment
+            @"<\s*comment-(start|end)[^>]*(>[^>]*<\/\s*comment-(start|end)[^>]*>|\/>)"
+        );
+
+        public static Release CreateAmendment(this Release release, DateTime createdDate, Guid createdByUserId)
+        {
+            var amendment = release.Clone();
+
+            // Set new values for fields that should be altered in the amended
+            // Release rather than copied from the original Release
+            amendment.Id = Guid.NewGuid();
+            amendment.Published = null;
+            amendment.PublishScheduled = null;
+            amendment.ApprovalStatus = ReleaseApprovalStatus.Draft;
+            amendment.Created = createdDate;
+            amendment.CreatedById = createdByUserId;
+            amendment.Version = release.Version + 1;
+            amendment.PreviousVersionId = release.Id;
+
+            var context = new Release.CloneContext(amendment);
+
+            amendment.Content = amendment
+                .Content
+                ?.Select(rcs => rcs.Clone(context))
+                .ToList();
+
+            amendment.ContentBlocks = amendment
+                .ContentBlocks
+                ?.Select(rcb => rcb.Clone(context))
+                .ToList();
+
+            amendment.RelatedInformation = amendment
+                .RelatedInformation
+                ?.Select(link => link.Clone())
+                .ToList();
+
+            amendment.Updates = amendment
+                .Updates
+                ?.Select(update => update.Clone(amendment))
+                .ToList();
+
+            UpdateAmendmentContent(context);
+
+            return amendment;
+        }
+
+        private static void UpdateAmendmentContent(Release.CloneContext context)
+        {
+            var replacements = new Dictionary<string, MatchEvaluator>();
+
+            // Bit cheeky to re-use the clone context, but it's a nice
+            // easy way to access and modify all of the content blocks
+            // that we used during the clone.
+            context.ContentBlocks
+                .ForEach(
+                    pair =>
+                    {
+                        switch (pair)
+                        {
+                            case { Key: DataBlock oldDataBlock, Value: DataBlock newDataBlock }:
+                                replacements[$"/fast-track/{oldDataBlock.Id}"] = _ => $"/fast-track/{newDataBlock.Id}";
+                                break;
+                        }
+                    }
+                );
+
+            var regex = new Regex(
+                string.Join('|', replacements.Keys.Append(FilterPattern)),
+                RegexOptions.Compiled
+            );
+
+            foreach (var contentBlock in context.ContentBlocks.Values)
```

Comment:
Will this have all the `HtmlBlocks` in it from the generic content sections? Does it have content blocks from the key stats, headlines, and summary sections to strip comments from them too?

I might be getting confused but just want to check as I think `Release.ContentBlock` only contains data blocks so wasn't sure if `context.ContentBlock` is the same.

Without running this and debugging it I'm not sure 🙁 .

Reply:
Yep it does. During the clone, we pass around the `CloneContext` which gets all of the content blocks appended to it. As mentioned in an earlier comment, it's a bit 'cheeky' to re-use it as it's intended for another purpose during the clone, however, it's definitely the easiest way to get all the right content block references that we need.

Reply:
Great, thanks for the explanation. I'm going to approve now.